message_feed: Improve tooltip on edited/moved notice.

somesh202 opened the pull request:
This PR modifies the default tooltip to a tippy one by removing the title attribute from the element and inserting the string text in aria-label attribute.
It uses the aria-label attribute to get the `last_edit_timestr` for the tooltip which is modified to follow the rules(english).
The tippy tooltip fetches the html content from a new `message_edit_notice_tooltip` template.
It also adds a second line to the tooltip: "Click to view edit history".
Currently on disabling the View edit history feature, the edited/moved notice gets hidden, so the idea of hiding the second line and showing only the first would not work in current scenario.

<!-- Describe your pull request here.-->
Fixes: #23075

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
![Screenshot from 2022-10-29 11-43-15](https://user-images.githubusercontent.com/77766761/198816871-ee06f281-3392-403f-8c02-da04d9a253fb.png)


**Self-review checklist**

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/version-control.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.

alya commented:
Thanks! Please leave a comment when you've had a chance to go through the PR checklist, and I'll take a look.

somesh202 commented:
Hi, I've gone through the checklist and also updated the PR description.

alya commented:
Looks good to me in manual testing! @sayamsamal could you please review this PR?

alya commented:
Oh, and this wasn't in the issue description, but I think we should italicize the second line, to match the styling we're implementing [here](https://github.com/zulip/zulip/pull/22746#issuecomment-1285017939).

somesh202 commented:
I've made the requested change and updated the screenshot.

sayamsamal commented:
Sure @alya, I'll review this PR.

sayamsamal commented on static/js/message_list_view.js:
```diff
@@ -257,11 +257,16 @@ export class MessageListView {
+            } else if (last_edit_date === "Yesterday") {
+                last_edit_date = "yesterday";
+            }
+            return last_edit_date + " at " + timerender.stringify_time(last_edit_time);
```
![image](https://user-images.githubusercontent.com/82862779/199787157-d047511f-9bc7-431a-8795-85d7087ed6bd.png)

@alya while showing actual dates, the requested format of Last {moved/edited} {date} sounds a bit odd, I think we need to add "on" before the date when we show the actual date.

alya replied:
Hm, I'm not sure -- I think it reads OK as-is, so I wonder if it's best to keep it simple (both in terms of the logic, and even more so for translation purposes). @timabbott what do you think?

timabbott replied:
I think it's OK as-is.

As a sidenote, using `+` to combine string fragments is not a correct way to write translatable strings; see 
https://zulip.readthedocs.io/en/latest/translating/internationalization.html for background. But generally one needs to have an entire sentence be a single translateable string.

somesh202 replied:
Here, instead of '+' , I constructed a single string using '`' template literals.

sayamsamal commented:
@somesh202 Good job on the PR. I have left you some reviews that need fixing. Everything else looks good to me, thanks for the PR!

sayamsamal commented:
FYI @alya, as mentioned by @somesh202, we do not display any text like "EDITED" or "MOVED" and also remove any such option from the message actions dropdown, when the organisation turns off message edit history, So hiding the Second Line won't provide us with any visual changes.

alya commented:
>  So hiding the Second Line won't provide us with any visual changes.

Ah, got it -- no need to make changes there, in that case.

somesh202 commented on static/templates/message_edit_notice_tooltip.hbs:
```diff
@@ -0,0 +1,7 @@
+
+<br/>
+
+{{#unless realm_allow_edit_history}}
```
After using `if` for the condition and passing `realm_allow_edit_history` in the tooltip content it works well, but i'm unable to fix the case when the notice still remains if we do not refresh the page. As, I can't find any connection between the enable history condition and the `message_edit_notice` class.

sayamsamal replied:
@somesh202 As the issue is with message edit notice not being removed until page refresh, it's not expected that your PR solved that part. As of the working of the tooltips, I went through your PR and everything seems to work great, thanks for the work.

@timabbott would have a better idea if we need to focus on updating the message edit notice (`MOVED`/`EDITED`) without having to refresh the page,

timabbott commented:
@somesh202 I see you pushed an updated version; can you explain how you addressed the review feedback? Replying to each review comment, plus adding a main comment with updated screenshots is usually the best practice.

somesh202 commented:
@timabbott Sorry for the late response, I've addressed the feedback and implemented the changes suggested by @sayamsamal and you, the UI for tooltip remains same . One issue still remains, i.e. the message edit notice is not removed until we refresh the page after changing the edit history settings, I'll try working this out too.

timabbott commented:
Closing as the original issue was fixed in https://github.com/zulip/zulip/pull/25883.
